rest_framework/fields.py

```diff
@@ -541,13 +542,25 @@ def run_validators(self, value):
         Test the given value against all the validators on the field,
         and either raise a `ValidationError` or simply return.
         """
+        from rest_framework.validators import ContextBasedValidator
+
         errors = []
         for validator in self.validators:
             if hasattr(validator, 'set_context'):
+                warnings.warn(
+                    "Method `set_context` on validators is deprecated and will "
+                    "no longer be called starting with 3.10. Instead derive the "
```

Comment:
This should be 3.11 now.

Reply:
btw - we've formalized our deprecation policy. This should use `RemovedInDRF312Warning`, which is currently a subclass of `PendingDeprecationWarning`. Deprecation classes can be found in `rest_framework/__init__.py`